openflow13/action.go

```diff
@@ -94,16 +93,87 @@ func DecodeAction(data []byte) Action {
 	case ActionType_SetNwTtl:
 		a = new(ActionNwTtl)
 	case ActionType_DecNwTtl:
-		a = new(ActionHeader)
+		a = new(ActionDecNwTtl)
 	case ActionType_SetField:
 		a = new(ActionSetField)
 	case ActionType_PushPbb:
 		a = new(ActionPush)
 	case ActionType_PopPbb:
 		a = new(ActionHeader)
+	case ActionType_Experimenter:
+		// For Experimenter message, there should be at least 8 bytes following action type field,
+		// including experimenter_id(4 byte), vendor(2 byte), and subtype(2 byte)
```

Comment:
Nope, binary.BigEndian.Uint16(data[8:]) would parse a uint16 number from 8th bit of data, and it would only consume 2 bytes, it is descided by Uint16.